### PR TITLE
ES7 Elastic server activator #7759

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch-server/src/test/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolverTest.java  1
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/ElasticsearchServerActivator.java  4
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/ElasticsearchServerSettings.java  4
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4739871&bid=15820800)

File: modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    public String resolveElasticServerDir()
+    {
+        final String value = configuration.esServerDir();
+
+        if ( !nullToEmpty( value ).isBlank() )
+        {
+            return value;
+        }
+
+        return HomeDir.get().toString() + "/../elasticsearch";
```
Comment (rymsha) [inline modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java]: Do not contact file paths.
add HomeDir.toFile().toPath() and resolve from it.

File: modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            clusterRoutingAllocationDiskThresholdEnabled( configuration.cluster_routing_allocation_disk_thresholdEnabled() ).
+            clusterName( configuration.cluster_name() ).
+            indexMaxResultWindow( configuration.index_maxResultWindow() ).
+            httpPort( configuration.http_port() ).
+            transportPort( configuration.transport_port() );
+
+        return builder.build();
+    }
+
+    public String resolveElasticServerDir()
```
Comment (rymsha) [inline modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java]: A lot of similar code above. Check for empty and resolve relative path could be in private function

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/ElasticsearchServerSettings.java  4
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java  5
- modules/core/core-elasticsearch-server/src/test/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolverTest.java  1
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/ElasticsearchServerActivator.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4739871&bid=15820800)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/ElasticsearchServerActivator.java  4
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolver.java  5
- modules/core/core-elasticsearch-server/src/main/java/com/enonic/xp/elasticsearch/server/impl/ElasticsearchServerSettings.java  4
- modules/core/core-elasticsearch-server/src/test/java/com/enonic/xp/elasticsearch/server/config/ElasticsearchServerConfigResolverTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4739871&bid=15820800)

**Pull request merged**